libr/anal/fcn.c

```diff
@@ -889,7 +884,29 @@ static void check_purity(HtUP *ht, RAnal *anal, RAnalFunction *fcn) {
 	r_list_free (refs);
 }
 
-static int fcn_recurse(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut8 *buf, ut64 len, int depth) {
+// TODO: move into io :?
+static int read_ahead(RAnal *anal, ut64 addr, ut8 *buf, int len) {
+	static ut8 cache[1024];
```

Comment:
:-1:

Reply:
its just a wrapper i find it very useful to test a varietyof things, i would keep it as long as it can be easily removed wiith 1 ifdef (READ_AHEAD). i know this must be moved into rio or rbuffer, but let me show what i find out with it:

* most reads are done forward
* some of those are backward (caching some bytes before the current read is useful)
* few reads happen far away
* no reads exceed 1KB
* r_io is pretty fast and it doesnt seems to impose a big slowdown, but caching speedups a bit
* ...